### PR TITLE
Enable printing of EXPLAIN query trees in the R client

### PR DESCRIPTION
**Pull request by rsund**
R client gave an error "Should not call dbFetch() on results that do not come from SELECT" for EXPLAIN queries. This pull request fixes that and implements a print method for showing the query trees in the R client.

### CODE REVIEW

**Review by Mytherin — APPROVED** [review]
Thanks for the PR! Looks good to me. Perhaps @krlmlr wants to have a look?

File: tools/rpkg/R/print__duckdb_explain.R [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+#' @rdname duckdb_explain-class
+#' @export
+duckdb_explain <- setClass("duckdb_explain", contains = "data.frame", slots = list(explain_key = "character", explain_value = "character"))
+
+#' @rdname duckdb_explain-class
+#' @usage NULL
+#' @export
+print.duckdb_explain <- function(x, ...) {
+  if (!all(names(x) %in% c("explain_key", "explain_value"))) NextMethod("print")
+  for (i in 1:nrow(x)) cat(paste0(x$explain_key[i], "\n", x$explain_value[i]))
```
Comment (krlmlr) [inline tools/rpkg/R/print__duckdb_explain.R:11]: Maybe:

```suggestion
  cat(paste0(x$explain_key, "\n", x$explain_value)
```

with a shortcut for length == 0?

**Review by krlmlr — COMMENTED** [review]
Thanks. Does this work with `dplyr::explain()` ?

File: tools/rpkg/R/print__duckdb_explain.R (hunk 1 above)
Comment (rsund) [reply tools/rpkg/R/print__duckdb_explain.R]: Thanks for the suggestion. Zero length output is now skipped. I also added `collapse=''` to` paste0()` to keep the output the same.

**Comment by rsund** [comment]
Thanks for the review. This works directly with the `dplyr::explain()` (i.e. with the default method in dbplyr). I will also add an option to select different plans with parameters in #2926 after this PR is merged.

**Comment by Mytherin** [comment]
Thanks!

File: tools/rpkg/R/print__duckdb_explain.R [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+#' @rdname duckdb_explain-class
+#' @export
+duckdb_explain <- setClass("duckdb_explain", contains = "data.frame", slots = list(explain_key = "character", explain_value = "character"))
+
+#' @rdname duckdb_explain-class
+#' @usage NULL
+#' @export
+print.duckdb_explain <- function(x, ...) {
+  if (!all(names(x) %in% c("explain_key", "explain_value"))) NextMethod("print")
+  if (nrow(x) > 0 & all(!is.na(x))) cat(paste0(x$explain_key, "\n", x$explain_value, collapse = ""))
```
Comment (krlmlr) [inline tools/rpkg/R/print__duckdb_explain.R:11]: What's the purpose of `all_(!is.na(x))` here?

Comment (rsund) [reply tools/rpkg/R/print__duckdb_explain.R:11]: There are two cases, when "zero" output occurs. Given that `x` is truly `NULL`, e.g. `x[FALSE,]`, that can be captured with `nrow(x)>0` and the out of "reasonable bounds" case such as `x[4,]` resulting in ugly printing of `NA`s  and can be captured using `all(!is.na(x))`. This escalates in situation like` x[4:50,]` and is better to silence.

**Pull request merged**